paddle/fluid/operators/mkldnn/mul_mkldnn_op.cc

```diff
@@ -0,0 +1,537 @@
+      const mkldnn::memory::desc& dst_desc,
+      const mkldnn::primitive_attr& fc_attr) {
+    const auto& fc_desc = inner_product_forward::desc(
+        prop_kind::forward, input_desc, weights_desc, dst_desc);
+
+    return inner_product_forward::primitive_desc(fc_desc, fc_attr,
+                                                 this->engine_);
+  }
+
+  mkldnn::memory CreateDstMemory(
+      const mkldnn::inner_product_forward::primitive_desc& fc_prim_desc,
+      const ExecutionContext& ctx, Tensor* output, bool force_fp32_output) {
+    auto dst_prim_desc = fc_prim_desc.dst_primitive_desc();
+    auto buffer_size = dst_prim_desc.get_size();
+
+    if (force_fp32_output) {
+      float* output_data = output->mutable_data<float>(
+          ctx.GetPlace(), ::paddle::memory::Allocator::kDefault, buffer_size);
+      output->set_format((memory::format)dst_prim_desc.desc().data.format);
+
+      return memory(dst_prim_desc, to_void_cast<float>(output_data));
+    } else {
+      int8_t* output_data = output->mutable_data<int8_t>(
+          ctx.GetPlace(), ::paddle::memory::Allocator::kDefault, buffer_size);
+      output->set_format((memory::format)dst_prim_desc.desc().data.format);
+
+      return memory(dst_prim_desc, to_void_cast<int8_t>(output_data));
+    }
+  }
+};
+
+static std::string GetHash(const Tensor* input, const Tensor* weights,
+                           const std::string& suffix) {
+  auto dim2str = [](const DDim& operand_dims) {
+    std::string str = "";
+    for (int i = 0; i < operand_dims.size(); ++i) {
+      str += std::to_string(operand_dims[i]) + "-";
+    }
+    return str;
+  };
+
+  std::string hash = std::to_string((unsigned)input->format()) +
+                     std::to_string((unsigned)input->type()) +
+                     dim2str(weights->dims()) + suffix;
+
+  return hash;
+}
+
+template <typename T, typename K>
+std::shared_ptr<MulPrimitiveFactory<T, K>> GetPrimitiveFactory(
+    const MKLDNNDeviceContext& dev_ctx, const ExecutionContext& ctx,
+    const Tensor* input, const Tensor* weights,
+    const mkldnn::engine& mkldnn_engine) {
+  const std::string key = GetHash(input, weights, ctx.op().Output("Out"));
+
+  auto prim_creator =
+      std::static_pointer_cast<MulPrimitiveFactory<T, K>>(dev_ctx.GetBlob(key));
+  if (prim_creator == nullptr) {
+    bool enable_quant =
+        std::is_same<T, int8_t>::value || std::is_same<T, uint8_t>::value;
+    if (enable_quant) {
+      prim_creator =
+          std::make_shared<QuantMulPrimitiveFactory<T, K>>(mkldnn_engine);
+    } else {
+      prim_creator = std::make_shared<MulPrimitiveFactory<T, K>>(mkldnn_engine);
+    }
+    dev_ctx.SetBlob(key, prim_creator);
+  }
+
+  return prim_creator;
+}
+
+template <typename T, typename K>
+class MulMKLDNNKernel : public framework::OpKernel<T> {
+ public:
+  void Compute(const framework::ExecutionContext& ctx) const override {
+    PADDLE_ENFORCE(paddle::platform::is_cpu_place(ctx.GetPlace()),
```

Comment:
you could remove paddle::

Reply:
okay, I'll fix it

Reply:
have you decided not to change this or not yet?

Reply:
Done! Please refer to ac7de0e